packages/ckeditor5-image/tests/image/utils.js

```diff
@@ -105,6 +107,79 @@ describe( 'image widget utils', () => {
 		} );
 	} );
 
+	describe( 'getImageWidgetAncestor()', () => {
+		let frag, caption;
+
+		describe( 'the selection is inside a caption', () => {
+			beforeEach( () => {
+				caption = writer.createContainerElement( 'figcaption' );
+				writer.insert( writer.createPositionAt( element, 1 ), caption );
+				frag = new ViewDocumentFragment( viewDocument, [ element ] );
```

Comment:
We use writer methods whenever possible and avoid class constructors. It is way safer.

`DowncastWriter.createDocumentFragment()`.